nuxt.config.ts

```diff
@@ -84,6 +89,17 @@ export default defineNuxtConfig({
     build: {
       target: 'esnext',
     },
+    optimizeDeps: {
```

Comment:
This change looks like a problem on the compile-time schedule, and probably a regression from Nuxt with the ability to create dependency between modules..

Reply:
Any change here is more likely a change in Vite (as we upgraded from Vite 4 -> Vite 5 in v3.9).

Reply:
I included the list for testing (testing with current main and this PR on my local), I need to remove all entries, we only need focus trap for client

Reply:
I think it is a good idea to include most of these in the config. We're lazy loading a lot of dependencies and they aren't discovered by the scanner. Some will end up triggering a full-reload on cold start. We may merge a new strategy in Vite 5.1 that we can adopt and this will be even more important then (we will start getting warnings to add the deps to the config). See:
- https://github.com/vitejs/vite/pull/15244

Reply:
has anyone in the team tried running dev in a clean clone (main branch with Nuxt 3.8)? there are almost 10 page refresh even when scrolling on timeline

@patak-dev I'm going to activate Vite 5 `{ server: { fs: cachedChecks: true } }` ;)

Reply:
@userquin see https://github.com/nuxt/nuxt/issues/24196#issuecomment-1853505843, there were issues in Nuxt 3.8 preventing the optimized deps cache to be used (so every startup was a cold start). This should be fixed in 3.9 (I imagine also in newer patches for 3.8?)

Reply:
I'm going to include also the new floating-vue v5.0.0 :fingers_crossed:

Reply:
@patak-dev I don't see any improvement activating `cachedChecks: true` on my laptop (with or without optimized deps, just a few ms, in both cases removing the `.cache` folder):

```shell
# without cachedChecks: true
ℹ Vite server warmed up in 8133ms                                                                                                                     15:02:18
ℹ Vite client warmed up in 12612ms                                                                                                                    15:02:19
✔ Nitro built in 5417 ms

# with cachedChecks: true
ℹ Vite server warmed up in 8190ms                                                                                                                     15:03:23
ℹ Vite client warmed up in 12484ms                                                                                                                    15:03:24
✔ Nitro built in 5196 ms